packages/backend/src/repositories/product.ts

```diff
@@ -125,4 +125,25 @@ export class ProductRepository {
       },
     });
   }
+
+  public async isActive(id: string, status: ProductStatus) {
+    return await this._dbClient.product.findFirst({
+      where: {
+        id,
+        status,
+      },
+    });
+  }
+
+  public async buy(id: string, userId: string, status: ProductStatus) {
+    return await this._dbClient.product.update({
+      where: {
+        id,
```

Comment:
question: can we save where { id, status: 'active }

Reply:
![image](https://user-images.githubusercontent.com/92884408/187428064-f67a5cd5-b389-4c4d-8517-68c305861736.png)
as i see, while update we can only search by id. There is no status parameter

Reply:
![image](https://user-images.githubusercontent.com/92884408/187435337-c303bcdb-86f4-4267-bdcc-404b1ab56ba0.png)
![image](https://user-images.githubusercontent.com/92884408/187435341-41313bd0-991a-4d6e-b672-e6c111b60f23.png)
If this solution it`s okay, i can commit it